runtime/sema/check_member_expression.go

```diff
@@ -112,17 +112,21 @@ func (checker *Checker) getReferenceType(typ Type, substituteAuthorization bool,
 }
 
 func shouldReturnReference(parentType, memberType Type) bool {
-	if memberType == nil || !isReferenceType(parentType) {
+	if memberType == nil {
+		return false
```

Comment:
Could you please add a test case covering this case?

Reply:
don't think `memberType` can be nil actually. Can remove this.

Reply:
Member type cannot be nil actually. Removed the nil check: c142c39b8556ec9812b186bbd30a3fbd7f60e633